tasks/firmware_installer.yml

```diff
@@ -10,7 +10,7 @@
   yum: name={{ item | splitext | first }}  state=latest
 
 - name: extract path for hpsetup command
-  shell: "rpm -ql {{ item }}|grep /hpsetup"
+  shell: "rpm -ql {{ item }}|grep -i 'hpsetup'"
```

Comment:
I looked through [1]  of the SPP yum manifests and found one file which also matches with this grep: 

<pre>
    <file>/usr/lib/x86_64-linux-gnu/hp-firmware-smartarray-46a4d957a7-8.32-1.1/.hpsetup</file>
    <file>/usr/lib/x86_64-linux-gnu/hp-firmware-smartarray-46a4d957a7-8.32-1.1/libhpsetup.so</file>
</pre>

Looks like it would still find the same directory as the ".hpsetup" for that specific firmware though.
Do you think we should make the regexp stricter to only match the ".hpsetup" and "/hpsetup"?
I have not looked through all the filelists for all the SPPs or fwrepos..

[1]: https://downloads.linux.hpe.com/SDR/repo/spp/2018.11.0/manifest/filelists.xml

Reply:
You're right. Some hosts have a .hpsetup file in this directory. This is strange. I've deleted the pull request.